File: Team_Workspace/dalia_ahmed/task10.py

class PaymentMethod:
    def pay(self, amount):
        print("Generic Payment")

class CreditCard(PaymentMethod):
    def pay(self, amount):
        print("Processing Credit Card: ",amount)

class Cash(PaymentMethod):
    def pay(self, amount):
        print("Accepting Cash: ",amount)



method=CreditCard()
method.pay(15.0)
method=Cash()
method.pay(15.0)

File: Team_Workspace/Fatma_Ahmed/cpp_tasks/task10/task10.cpp
// task10.cpp : This file contains the 'main' function. Program execution begins and ends there.
//

#include <iostream>
using namespace std;



class PaymentMethod {
public:
    virtual void pay(double amount)
    {
        cout << "Generic Payment Processing ....\n";

    }
};

class CreditCard : public PaymentMethod {
public:

    void pay(double amount) override {
        cout << "Processing Credit Card payment of " << amount << endl;

    }
};

class Cash :public PaymentMethod {
public:
    void pay(double amount) override
    {
        cout << "Accepting Cash payment of " << amount << endl;

    }
};

int main()
{
    CreditCard myCard;
    PaymentMethod& paymentSystem = myCard;

    paymentSystem.pay(15.50);

}

